packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { css } from '@emotion/core';
+import { XOR } from '../../utils';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { Button, ButtonProps, ButtonSize } from '../Button';
+import { ButtonColor } from '../Button';
+import { IconButton } from '../IconButton';
+import { IconProps } from 'react-magma-icons';
+import {
+  ToggleButtonGroupContext,
+  setButtonStyles,
+  setIconWidth,
+} from '../ToggleButtonGroup/ToggleButtonGroup';
+
+export interface ToggleButtonTextProps extends ButtonProps {
+  /**
+   * Enables a specified button to be active.
+   */
+  defaultChecked?: boolean;
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  isChecked?: boolean;
```

Comment:
Still a little confused as to why we need both `defaultChecked` and `isChecked` 🤔 I think ideally we figure out a way to combine these two and think through how adopters will use this.